### ISSUE TITLE
Allow to append files to the latest release, overwriting existing files

### ISSUE THREAD

**Issue by XapaJIaMnu** [marvinpinto/actions#177]
Hey,

I have a project that I use github CI to automatically build and I also want to rolling-release the latest git master.

My problem is that I have several different workflows: Ubuntu{16-18-20}, MacOS and Windows and I want to release binaries for all of them. Unfortunately at the moment, I don't see a way to append to an existing release. The old release is deleted and a new one is created, which makes it so that only the workflow that finishes the last, would have its artefact uploaded.

Is there a way to solve that?

Cheers,

Nick

**Comment by pojntfx** [comment]
I would also very much so appreciate this feature; a usecase I could see this to be very useful in is having a build matrix and each runner uploading the artifacts independendly.

**Comment by mgree** [comment]
I agree, this would be a very nice feature! A workaround I've found is to use different `latest` tags for different OSes. My [prerelease `latest` action](https://github.com/mgree/ffs/blob/main/.github/workflows/latest.yml) uses `${{ matrix.os }}` as the `automatic_release_tag`. Conventiently, I'm building on `ubuntu-latest` and `macos-latest`, so the names look pretty good as release tags!  I'm not certain exactly how this would work if your build matrix has multiple versions of a given OS, like the OP mentioned.

I tried two other approaches, neither of which worked.

- [Using `$RUNNER_OS`](https://github.com/mgree/ffs/blob/b5090fdefd43101e5d53cd95c9a853704e342437/.github/workflows/latest.yml#L43-L51): the environment variable never got subbed in. (Could be my bad.)
- [Renaming the executable](https://github.com/mgree/ffs/blob/81fcbf5a5e155aab059d6bdf74a788c1680b65c9/.github/workflows/latest.yml#L43-L62): the last run to finish overwrote the others, as the OP notes.

**Comment by devinstasafe** [comment]
The reason it does not work is because it is [actually deleting the older release](https://github.com/marvinpinto/actions/blob/c86cacd289b8391d77f267b976e830c25826e933/packages/automatic-releases/src/main.ts#L298) (with same tag) and creating a new one.

If a flag is added to "update" or "edit" then it would be ideal. I have the same use case

**Comment by mgree** [comment]
Update or edit flags sound good, but then users of the action will need to be careful: it seems like you can't have more than one file with the same name in a GitHub release.

I think a useful mode would be "append", where no existing files in the release should be altered---and there's a warning/failure if they are. To be most helpful, the warning should give a hint about renaming files to be unique.

**Comment by mgree** [comment]
I found an even better workaround. You can see the whole thing in my unified [build.yml for testing, benchmarking, and prereleasing](https://github.com/mgree/ffs/blob/66a104671f9e5f827053f7760d912234c0c8fc23/.github/workflows/build.yml), but here's the general idea:

  1. In the job that compiles your code, generate uniquely named artifacts on each architecture you care about. I use `if`s, but a well-placed `format` might let you have fewer steps. ([See relevant bit of code.](https://github.com/mgree/ffs/blob/66a104671f9e5f827053f7760d912234c0c8fc23/.github/workflows/build.yml#L54-L66))
  2. Set up another job for prereleasing that [depends on the prior one](https://github.com/mgree/ffs/blob/66a104671f9e5f827053f7760d912234c0c8fc23/.github/workflows/build.yml#L83) and runs [when you want to do a prerelease](https://github.com/mgree/ffs/blob/66a104671f9e5f827053f7760d912234c0c8fc23/.github/workflows/build.yml#L85). While you might be compiling on many architectures, this can run on whatever; I use `ubuntu-latest` because it seems to have the fastest startup time. ([See relevant bit of code.](https://github.com/mgree/ffs/blob/66a104671f9e5f827053f7760d912234c0c8fc23/.github/workflows/build.yml#L82-L85))
  3. Download all of the binaries and give them appropriate names. ([See the relevant bit of code.](https://github.com/mgree/ffs/blob/66a104671f9e5f827053f7760d912234c0c8fc23/.github/workflows/build.yml#L87-L95))
  4. Deploy the prerelease with all of the files together. ([See the relevant bit of code.](https://github.com/mgree/ffs/blob/66a104671f9e5f827053f7760d912234c0c8fc23/.github/workflows/build.yml#L97-L106))

It definitely took some experimentation to get the names all correct, and some of the component workflows will fail to upload things and still report a green status.

Hope this helps!

**Comment by triplef** [comment]
@mgree this is a great solution, thank you.

One thing to look out for is how actions/download-artifact stores the artifacts in individual subfolders, so I used a `**` glob for `files` to find the artifacts. [Here is](https://github.com/gnustep/tools-android/blob/95686ccc2bbd9261a565d4269f2695880b858000/.github/workflows/ci.yml#L50-L73) what I ended up with.
